packages/cli/src/code-format.ts

```diff
@@ -0,0 +1,187 @@
+        default: true,
+    })
+    .option('debug', {
+        type: 'boolean',
+        description: 'Enable explicit debug log',
+        alias: 'D',
+        default: false,
+    })
+    .option('require', {
+        type: 'array',
+        description: 'require hooks',
+        alias: 'r',
+        default: [] as string[],
+    })
+
+    .alias('h', 'help')
+    .alias('v', 'version')
+    .help()
+    .strict()
+    .wrap(yargs.terminalWidth())
+    .parseSync();
+
+const {
+    debug,
+    endWithNewline,
+    indentEmptyLines,
+    indentSize,
+    indentWithTabs,
+    maxPerserveNewlines,
+    newlineBetweenRules,
+    perserveNewlines,
+    require: requires,
+    selectorSeparatorNewline,
+    target,
+} = argv;
+
+const log = createLogger('[Stylable code formatter]', debug);
+
+log('[Arguments]', argv);
+
+// execute all require hooks before running the CLI build
+for (const request of requires) {
+    if (request) {
+        require(request);
+    }
+}
+
+const stylable = Stylable.create({
+    fileSystem: nodeFs,
+    requireModule: require,
+    projectRoot: target,
+    resolverCache: new Map(),
+});
+const lsp = new StylableLanguageService({ fs: nodeFs, stylable });
+
+function readDirectoryDeep(dirPath: string, fileSuffixFilter = '.st.css') {
+    const files = nodeFs.readdirSync(dirPath, 'utf-8');
+    let res: string[] = [];
+
+    for (const item of files) {
+        const currentlFilePath = join(dirPath, item);
+        const itemStat = nodeFs.statSync(join(currentlFilePath));
+
+        if (itemStat.isFile() && item.endsWith(fileSuffixFilter)) {
+            res.push(currentlFilePath);
+        } else if (itemStat.isDirectory()) {
+            res = res.concat(readDirectoryDeep(currentlFilePath));
+        }
+    }
+
+    return res;
+}
+
+function formatStylesheet(filePath: string) {
+    log('Formatting: ' + filePath);
+    const fileContent = nodeFs.readFileSync(filePath, 'utf-8');
+
+    const formatting = lsp.formatDocument(
+        filePath,
+        { start: 0, end: fileContent.length },
+        {
+            end_with_newline: endWithNewline,
+            indent_empty_lines: indentEmptyLines,
+            indent_size: indentSize,
+            indent_with_tabs: indentWithTabs,
+            max_preserve_newlines: maxPerserveNewlines,
+            newline_between_rules: newlineBetweenRules,
+            preserve_newlines: perserveNewlines,
+            selector_separator_newline: selectorSeparatorNewline,
+        }
+    );
+
+    if (formatting.length) {
+        writeFileSync(filePath, formatting[0].newText);
+        log('File formatted successfully');
+    } else {
+        log('No formatting required');
+    }
+}
+
+log('Starting code formatting process');
+const formatPathStats = nodeFs.statSync(target);
+
+if (formatPathStats.isFile()) {
+    if (target.endsWith('.st.css')) {
+        formatStylesheet(target);
+    } else {
+        throw new Error('cannot format file, not a Stylable stylesheet (.st.css)');
```

Comment:
maybe support normal css files as well?

Reply:
It passes through our LSP which sanitizes and sets up all sorts of things. The user would probably be better off using `js-beautify` directly in their project.